src/sipsess/connect.c

```diff
@@ -165,12 +165,6 @@ static void invite_resp_handler(int err, const struct sip_msg *msg, void *arg)
 		if (sess->redirecth)
 			sess->redirecth(msg, sip_dialog_uri(sess->dlg),
 				        sess->arg);
-
-		err = invite(sess);
-		if (err)
-			goto out;
-
-		return;
```

Comment:
please add a new boolean flag, e.g. "bool auto_follow_redirect" and make it default true.

then add setter functions to enable/disable it.

Reply:
Isn't it enough that the app decides on how to react to 3xx response? Why would the decision be done already in re?

Reply:
with this change, we have to upgrade both re and app (e.g. baresip) at the same time,
which is not convenient.

We should try to keep backwards compatibility so that the library works as before.

over time we can phase out the behaviour...